Implementation of multiple results per match.

citron0xa9 opened the pull request.

citron0xa9 committed:
refs #99: Implementation of multiple results per match.

 - results can be added and removed dynamically on the edit match page

nikriek commented on spec/models/game_result_spec.rb:
```diff
@@ -0,0 +1,26 @@
+RSpec.describe GameResult, type: :model do
+  it "is valid when produced by a factory" do
+    result = FactoryBot.build(:game_result)
+    result.save
```
You dont need to save the result. Make it a one line liek

```
expect(FactoryBot.build(:game_result)).not_to be_valid
```

nikriek commented on spec/models/game_result_spec.rb:
```diff
@@ -0,0 +1,26 @@
+  end
+
+  it "fails validation with non-number scores" do
+    result = FactoryBot.build(:game_result, score_home: "2a")
```
You might also check for the existence of a `score_home` in the errors dict after calling `is_valid?`

```ruby
r = FactoryBot.build(:game_result, score_home: "2a")
expect {r.is_valid?}.to change { r.errors }.from([]).to([score_home: [...]])
# OR
expect {r.is_valid?}.to change { r.errors[:score_home].count }.from(0).to(1)
```

nikriek commented on spec/models/game_result_spec.rb:
```diff
@@ -0,0 +1,26 @@
+
+  it "is valid without any scores given" do
+    result = GameResult.new
+    match = FactoryBot.create(:match)
```
just pass it directly in the constructor and check for validity

nikriek commented on spec/factories/matches.rb:
```diff
@@ -22,10 +22,15 @@
     points_away 3
     points_home 1
+
+    factory :match_with_results do
```
Have a look at rspec traits

citron0xa9 replied:
I think this is not doable because the callback `after(:create)` is used

nikriek reviewed (approved):
Looks good. just minor stuff

citron0xa9 committed:
Merge remote-tracking branch 'origin/dev' into dividebyzero/#99/multiple_results_per_match

citron0xa9 committed:
refs #99: Applied feedback

citron0xa9 committed:
Merge remote-tracking branch 'origin/dev' into dividebyzero/#99/multiple_results_per_match

citron0xa9 committed:
refs #99: Improved tests, fixed some errors in match model

citron0xa9 committed:
Merge remote-tracking branch 'origin/dev' into dividebyzero/#99/multiple_results_per_match

citron0xa9 committed:
refs #99: Improvements for PR, changed dependence between match and game results

citron0xa9 committed:
refs #99: Added no-script functionality

citron0xa9 committed:
refs #99: Further improved noscript functionality, added localization and tests

citron0xa9 committed:
refs #99: Fixed match updating problems

 - Moved entering game results into extra view because otherwise
other attributes of matches got wrongly updated
 - Added needed tests and localization
 - Adjusted tests

citron0xa9 committed:
Merge remote-tracking branch 'origin/dev' into dividebyzero/#99/multiple_results_per_match
